File: 1647 - Minimum Deletions to Make Character Frequencies Unique/solution.cpp
class Solution {
public:
    int minDeletions(string s) {
        int res = 0;
        vector<int> cnt(26);
        unordered_set<int> seen;
        for(auto c : s) cnt[c-'a']++;
        for(auto i : cnt){
            while(i>0 && seen.count(i)) res++, i--;
            seen.insert(i);
        }
        return res;
    }
};

File: 1647 - Minimum Deletions to Make Character Frequencies Unique/solution.py
from collections import Counter
class Solution:
    def minDeletions(self, s: str) -> int:
        res, cnt, seen = 0, Counter(s), set()
        for i in cnt.values():
            while i > 0 and i in seen:
                res += 1; i -= 1
            seen.add(i)
        return res